test/blackbox/common/DDSBlackboxTestsPersistenceGuid.cpp

```diff
@@ -186,6 +186,169 @@ TEST_P(PersistenceGuid, SetPersistenceGuidByXML)
+ * @brief This test checks if the persistence desn't work for TRANSIENT_LOCAL endpoints when it is not used the option
+ * 'dds.persistence.also-support-transient-local'.
+ *
+ * This test creates a DataReader and DataWriter configuring for each of them the persistence plugin to use SQLITE3,
+ * the database filename, a specific persistence guid, and the TRANSIENT_LOCAL durability and reliability QoS.
+ * Once both are discovered, it generates a new data message to send, since every time a new message is sent the middleware doesn't insert
+ * a new entry on the persistence database.
+ * In order to check if the persistence guid specified is not applied correctly, the test uses the python file 'check_guid.py'
+ * that returns the number of apparitions of a guid, on a specific database and table.
+ */
+TEST_P(PersistenceGuid, NoSetPersistenceForTransientLocal)
+{
+    // Configure Participant Properties
+    PropertyPolicy participant_policy;
+    participant_policy.properties().emplace_back("dds.persistence.plugin", "builtin.SQLITE3");
+    participant_policy.properties().emplace_back("dds.persistence.sqlite3.filename", "persistence.db");
+
+    // Configure Writer Properties
+    PropertyPolicy writer_policy;
+    writer_policy.properties().emplace_back("dds.persistence.guid", "77.72.69.74.65.72.5f.70.65.72.73.5f|67.75.69.64");
+
+    // Create DataWriter and configure the durability and reliability QoS
+    PubSubWriter<HelloWorldType> writer(TEST_TOPIC_NAME);
+    writer.property_policy(participant_policy);
+    writer.entity_property_policy(writer_policy);
+    writer.durability_kind(TRANSIENT_LOCAL_DURABILITY_QOS);
+    writer.reliability(RELIABLE_RELIABILITY_QOS);
+
+    writer.init();
+    ASSERT_TRUE(writer.isInitialized());
+
+    // Configure Reader Properties
+    PropertyPolicy reader_policy;
+    reader_policy.properties().emplace_back("dds.persistence.guid", "77.65.61.64.65.72.5f.70.65.72.73.5f|68.76.70.65");
+
+    // Create DataReader and configure the durability and reliability QoS
+    PubSubReader<HelloWorldType> reader(TEST_TOPIC_NAME);
+    reader.property_policy(participant_policy);
+    reader.entity_property_policy(reader_policy);
+    reader.durability_kind(TRANSIENT_LOCAL_DURABILITY_QOS);
+    reader.reliability(RELIABLE_RELIABILITY_QOS);
+
+    reader.init();
+    ASSERT_TRUE(reader.isInitialized());
+
+    // Wait until both of them are discovered
+    reader.wait_discovery();
+    writer.wait_discovery();
+
+    // Generate a new data message to send
+    auto data = default_helloworld_data_generator(1);
+    auto aux = data;
+    writer.send(data);
+    reader.startReception(aux);
+    reader.block_for_all();
+#ifdef WIN32
+    // Check if there is one entry in the writers database table with the stated persistence guid
+    int result1 = system(
+        "python check_guid.py \"persistence.db\" \"writers\" \"77.72.69.74.65.72.5f.70.65.72.73.5f|67.75.69.64\"");
+    ASSERT_EQ(result1, 1);
```

Comment:
This test is failing in Windows. I think it should be 
```suggestion
    ASSERT_EQ(result1, 255);
```